### PR TITLE
Add tparallel linter

### LINKED ISSUE #1371: Add support for "tparallel"
[tparallel](https://github.com/moricho/tparallel) finds inappropriate or inefficient usage of `t.Parallel()` method in Go test codes.
It detects the following:
- insufficient parallelization: `t.Parallel()` is called in either a top-level test function or a sub-test function only
- inappropriate teardown process: Although `t.Parallel()` is called in the sub-test function, it is post-processed by `defer` instead of `t.Cleanup()`

Particularly in tests that access the database, the communication time to the database takes up a large portion of the test time, so we can reduce the test time by executing the test code in parallel appropriately.

### PR DESCRIPTION
**Pull request by Nivl**
This PR adds the [tparallel](https://github.com/moricho/tparallel) linter. This linter checks for inefficient usage of `t.Parallel()` in the tests.

Closes #1371 (cc @moricho)

~The Nancy failure is being taken care of in #1378 and #1293~ Merged

~Note: After doing more manual tests with the linter, I found a false positive: https://github.com/moricho/tparallel/issues/8. That shouldn't prevent reviewing though, I'll update the linter's version once it's been fixed.~ Done

### CODE REVIEW

File: test/linters_test.go [hunk 1]
```diff
@@ -252,11 +252,55 @@ func TestExtractRunContextFromComments(t *testing.T) {
...
+		cfg, err := yaml.Marshal(rc.config)
+		assert.NoError(t, err)
+
+		testshared.NewLintRunner(t).RunWithYamlConfig(string(cfg), args...).
+			ExpectHasIssue(
+				"testdata/tparallel/missing_subtest_test.go:7:6: TestSubtests's subtests should call t.Parallel\n",
+			)
+	})
+
+	t.Run("should pass on parallel test with no subtests", func(t *testing.T) {
```
Comment (Nivl) [inline test/linters_test.go:292]: This test will fail because of a false-positive. See https://github.com/moricho/tparallel/issues/8

Comment (moricho) [reply test/linters_test.go:292]: I fixed that bug and released a new version.
https://github.com/moricho/tparallel/releases/tag/v0.2.1

Comment (Nivl) [reply test/linters_test.go:292]: Awesome, thanks

**Comment by Nivl** [comment]
@golangci/team up :)

**Review by saschagrunert — APPROVED** [review]
LGTM

**Pull request merged**